src/api/background/handlers/browser/window-close/window-close.handler.ts

```diff
@@ -0,0 +1,20 @@
+import { removeDecryptionKey } from "~wallets/auth";
+import browser from "webextension-polyfill";
+
+/**
+ * Listener for browser close.
+ * On browser closed, we remove the
+ * decryptionKey.
+ */
+export async function handleWindowClose() {
+  const windows = await browser.windows.getAll();
+
+  // TODO: Maybe we should be counting connected apps instead?
```

Comment:
The logic here is probably incorrect. We might want to open a separate issue to address this.

Reply:
Yes, let's create a separate issue for this 👍